### ISSUE TITLE
Returning a response is forcing application/json content-type

### ISSUE THREAD

**Issue by marbemac** [elysiajs/elysia#827]
### What version of Elysia is running?

1.1.12

### What steps can reproduce the bug?

I'm running a Vite dev server with bun + elysia, not sure if that is relevant BUT the issue is pretty strait forward:

My handler has code along these lines:

```ts
const r = new Response(html, { headers: { 'content-type': 'text/html' } });

// This logs out "_Response", which turns out to be important due to internal elysia handler logic
console.log(r?.constructor?.name);

return r;
```

Seems like response mapping depends on the constructor name property, and checks for "Response" here -> https://github.com/elysiajs/elysia/blob/main/src/handler.ts#L471.

This check does not pass, since in my setup it turns out to be "_Response". 

Thus, it falls back to a instanceof check in the default handler, which does pass -> https://github.com/elysiajs/elysia/blob/main/src/handler.ts#L582-L587.

Problem is that logic doesn't return the response as-is - it forces an application/json content-type. Might make sense to pass response as-is and only set the content-type if not already present?

**Comment by SaltyAom** [comment]
Fixed and published under [1.1.13](https://github.com/elysiajs/elysia/releases/tag/1.1.13) with test case included
see https://github.com/elysiajs/elysia/compare/1.1.12...1.1.13

If this doesn't fix the issue, feel free to tag me or reopen the issue.

**Comment by marbemac** [comment]
Nice, the changes look like it should resolve it for sure, thanks!
